Source/WebKit/WebProcess/WebCoreSupport/WebDataListSuggestionPicker.h

```diff
@@ -47,14 +40,23 @@ namespace WebKit {
 
 class WebPage;
 
-class WebDataListSuggestionPicker final : public WebCore::DataListSuggestionPicker {
+class WebDataListSuggestionPicker final : public RefCounted<WebDataListSuggestionPicker>, public WebCore::DataListSuggestionPicker {
+    WTF_MAKE_TZONE_ALLOCATED_INLINE(WebDataListSuggestionPicker);
 public:
-    WebDataListSuggestionPicker(WebPage&, WebCore::DataListSuggestionsClient&);
+    static Ref<WebDataListSuggestionPicker> create(WebPage& page, WebCore::DataListSuggestionsClient& client)
+    {
+        return adoptRef(*new WebDataListSuggestionPicker(page, client));
+    }
+
+    void ref() const final { RefCounted::ref(); }
```

Comment:
Ouch, I think this actually introduces a security bug:
```
    WebCore::DataListSuggestionsClient& m_client;
    WeakRef<WebPage> m_page;
```

`m_client` can go stale now :(